steps/data_embedding_steps/compute_embedding_drift_step/compute_embedding_drift_step.py

```diff
@@ -0,0 +1,154 @@
+    reference_data_version: str, current_data_version: str, distance: float
+) -> Dict[str, Union[str, float, bool]]:
+    """Construct a payload for send the embedding drift data to the metric service via post request.
+
+    Args:
+        reference_data_version (str): the version identifier for the reference data.
+        current_data_version (str): the version identifier for the current data.
+        distance (float): the computed Euclidean distance between the embeddings of the reference and current data.
+
+    Returns:
+        Dict[str, Union[str, float, bool]]: a dictionary containing the 4 items required by the metric service embedding drift relation.
+    """
+    drifted = distance > 0
+
+    formatted_reference_data_version = f"'{reference_data_version}'"
+    formatted_current_data_version = f"'{current_data_version}'"
+
+    return {
+        "reference_dataset": f"{formatted_reference_data_version}",
+        "current_dataset": f"{formatted_current_data_version}",
+        "distance": distance,
+        "drifted": drifted,
+    }
+
+
+@step
+def compute_embedding_drift(
+    collection_name: str, reference_data_version: str, current_data_version: str
+) -> float:
+    """Compute the measure of 'drift' in data embeddings between the current and reference datasets, identified by the given collection name.
+
+    This function calculates the Euclidean distance between the mean values of the reference and current embeddings
+    This distance signifies the 'drift' or variation in the data distribution between the reference and current datasets, which will be visualised over time using a plot of the distance
+
+    Args:
+        collection_name (str): the name of the collection to compute
+        reference_data_version (str): the reference data version
+        current_data_version (str): the current data version
+
+    Returns:
+        float: the Euclidean distance representing the drift between the reference and current datasets. 0 if reference and current embeddings are the same.
+    """
+    # Create a chromadb client
+    # Switch hostname to chroma-service.default if running the pipeline on k8s
+    chroma_client = ChromaStore(
+        chroma_server_hostname="localhost", chroma_server_port=8000
+    )
+    (
+        reference_embeddings,
+        current_embeddings,
+    ) = chroma_client.fetch_reference_and_current_embeddings(
+        collection_name, reference_data_version, current_data_version
+    )
+    validate_embeddings(reference_embeddings, current_embeddings)
+    distance = calculate_euclidean_distance(reference_embeddings, current_embeddings)
+
+    logger.info(
+        f"The Euclidean distance between the mean of reference embeddings and the mean of current embeddings is: {distance}"
+    )
+
+    payload = build_embedding_drift_payload(
+        reference_data_version, current_data_version, distance
+    )
+    response = requests.post("http://localhost:5000/embedding_drift", json=payload)
```

Comment:
`localhost` and `5000` should be a variable e.g. `MONITORING_METRICS_HOST_NAME` and `MONITORING_METRICS_PORT`. If this pipeline runs on k8s, the `localhost` will not be reachable so, it should be highlighted that the name of hostname will be `monitoring-service.default` when running the pipeline on k8.

Reply:
I updated readme so that people will need to port forward the monitoring service. `monitoring-service.default` will only work when pipeline is also run on k8s right?

Reply:
Correct, in addition the port forward should be done in a separate terminal. Having what value of host name to use as when running on k8s commented would be useful, same as chroma has mentioned.